addons/account/models/account_order_mixin.py

```diff
@@ -0,0 +1,230 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, _, api, fields, models
+from odoo.tools import float_is_zero
+
+
+class AccountOrderMixin(models.AbstractModel):
+    # This class enables downpayment support for models that can have down payments on them
+    _name = 'account.order.mixin'
+    _description = 'Account Order Mixin'
+
+    amount_to_invoice = fields.Monetary("Amount to invoice", store=True, compute='_compute_amount_to_invoice')
+    amount_invoiced = fields.Monetary("Already invoiced", compute='_compute_amount_invoiced')
+
+    # To override
+    company_id = fields.Many2one('res.company')
+    currency_id = fields.Many2one('res.currency')
+    fiscal_position_id = fields.Many2one('account.fiscal.position', string='Fiscal Position')
+    partner_id = fields.Many2one('res.partner')
+
+    amount_total = fields.Monetary()
+
+    invoice_ids = fields.Many2many('account.move')
+    order_line = fields.One2many('account.order.line.mixin')
+
+    invoice_status = fields.Selection([])
```

Comment:
The common values for the selection could probably be defined here, just for clarity. You'll change some labels in the override anyway

Reply:
Yeah they have some common values indeed but they have different labels (ex. 'nothing to bill' vs 'nothing to invoice'), so I'm not sure it is possible to define them on the mixin?

Reply:
You can define the common things on the mixin, and only the label in each of the modules :)